agent/consul/gateway_locator.go

```diff
@@ -74,9 +75,19 @@ type GatewayLocator struct {
 // our chosen mesh-gateway configuration can reach the primary's servers (like
 // a ping or status RPC) we cheat and use the federation state replicator
 // goroutine's success or failure as a proxy.
-func (g *GatewayLocator) SetLastFederationStateReplicationError(err error) {
+func (g *GatewayLocator) SetLastFederationStateReplicationError(err error, fromReplication bool) {
+	if g == nil {
+		return
+	}
+
 	g.lastReplLock.Lock()
 	defer g.lastReplLock.Unlock()
+
+	if fromReplication {
+		// If we get info from replication, assume replication is operating.
+		g.useReplicationSignal = true
```

Comment:
What do we mean by replication is operating? Just that there's a Replicator routine running?

Where I'm lost is that there could have been an error in this call to `fetchRemote`:
https://github.com/hashicorp/consul/pull/9366/files#diff-1b7bee896bb2362c8062a2d6ee0f449799817d428f000c8a5c6347e5a041f59aR42

Reply:
Sigh. While on the circular quest to fix this series of wanfed adverse interactions I completely missed that the `err` coming into this function can be `nil` when plumbing this up.

Reply:
It's supposed to be "successfully did a remote RPC"

Reply:
Fixed.